lua/damagelogs/shared/autoslay.lua

```diff
@@ -374,29 +376,27 @@ if CLIENT then
         end
 
         for _, v in pairs(walls) do
-            table.insert(ents, v)
+            table.insert(jails, v)
         end
     end)
 
-    hook.Add("Think", "JailWalls", function()
-        local function CheckWalls()
-            local found = false
-
-            for k, v in pairs(ents) do
-                if IsValid(v) then
-                    v:SetCustomCollisionCheck(true)
-                    v.jailWall = true
-                    table.remove(ents, k)
-                    found = true
-                    break
-                end
-            end
+    local function CheckWalls()
+        local found = false
 
-            if found then
-                CheckWalls()
+        for k, v in ipairs(jails) do
+            if IsValid(v) then
+                v:SetCustomCollisionCheck(true)
+                v.jailWall = true
+                table.remove(jails, k)
+                found = true
+                break
             end
         end
 
-        CheckWalls()
-    end)
+        if found then
+            CheckWalls()
```

Comment:
I know you didn't write this, but I wonder why this function calls itself recursively, instead of just continuing the for loop?

Reply:
I'm glad you asked that question. I don't see any reason why not.